doc/vxlan/data_plane_VXLAN_HLD.md

```diff
@@ -0,0 +1,507 @@
+
+# DATA PLANE VXLAN HLD
+
+## Table of Content
+
+- [Revision](#revision)
+- [Definition/Abbreviation](#definitionsabbreviations)
+- [Overview](#overview)
+- [Use case](#use-case)
+- [Architecture](#architecture-design)
+- [Configuration and management](#2-configuration-and-management)
+- [Config DB Enhancements](#config-db-enhancements)
+- [APP_DB Enhancements](#3-app_db-enhancements)
+- [Modules Design and Flows](#4-modules-design-and-flows)
+- [Tunnel creation](#43-tunnel-creation)
+- [MAC learning flow](#figure-3-mac-learning-flow)
+
+## Revision
+
+|Rev | Date	 | Author       | Change Description |
+|----|-------|--------------|--------------------|
+|1.0 |       | Vladimir Kuk | Initial version    | 
+
+
+### Scope
+
+This document provides general information about the Data plane VXLAN feature implementation based on RFC 7348 in SONiC.
+This document provides a high-level design (HLD) for supporting Data plane VXLAN configuration in SONiC. The aim is to enable users to manually configure VXLAN tunnels, bypassing the need for dynamic discovery protocols such as EVPN. This approach allows for simpler deployment scenarios and can be beneficial in environments where dynamic protocols are not preferred or supported.
+
+
+### Definitions/Abbreviations
+
+#### Table 1: Abbreviations
+
+| Term  | Meaning                                   |
+|-------|-------------------------------------------|
+|VXLAN	|Virtual Extended LAN                       |
+|VNI    |Vxlan Network Identifier                   |
+|VTEP   |Vxlan Tunnel End Point                     |
+|EVPN   |Ethernet Virtual Private Network           |
+|NVO    |Network Virtualization Overlay             |
+|VM     |Virtual Machine                            |
+|BUM	|Broadcast, Unknown unicast and Multicast   |
+|IMET	|Inclusive Multicast Ethernet Tag           |
+|P2P	|Point to Point Tunnel                      |
+|P2MP	|Point to MultiPoint Tunnel                 |
+|BGP	|Border Gateway Protocol                    |
+
+
+### Overview
+
+Current SONiC implementation supports BGP-EVPN for VXLAN tunnel which provide auto-discovering of remote VTEPs, dynamic create and removal of tunnels, dynamic VLAN extensions, and so on. This however comes with complexity of setting BGP configuration, as well as high resource requirements, which are not necessarily warranted in smaller networks.
+
+In order to simplify the configuration and reduce memory consumption, data plane L2 learning can be used instead.
+The main differences between two approaches are that L2 learning is enabled on tunnel interface and all of the configuration is static.
+
+### Requirements
+
+* Static VXLAN Tunnel Configuration: Users should be able to manually specify VXLAN tunnel endpoints.
+* VNI to VLAN Mapping: Users must be able to statically map VNIs to VLANs.
+* L2 learning over local and remote VTEPs.
```

Comment:
This HLD supports full dynamic learning, especially on remote VTEPs.
The purpose of this HLD wasn't replace BGP with controller. That is why we don't support MAC pointing to static VXLAN.
If we want to support command to configure static MAC with move option, in case of actual mac move, this command will need to be deleted from Config db (not sure if this is acceptable behavior in SONiC).